### PR TITLE
do not install docker with curl

### LINKED ISSUE #109577: Curl pipe to shell for docker in e2e tests setup
### What happened?

setup_host.sh in e2e_node test curls https://get.docker.com and directly pipes to shell:

https://github.com/kubernetes/kubernetes/blob/f173d01c011c3574dea73a6fa3e20b0ab94531bb/test/e2e_node/environment/setup_host.sh#L82-L84

The issue is that this may present a supply chain security risk, as we are directly executing the script from get.docker.com.

### What did you expect to happen?

I would instead expect one of the following:

1. The host includes docker already via it's OS image
2. The script is checked into this repo and version controlled
3. The script is at least verified to be unchanged via hash

### How can we reproduce it (as minimally and precisely as possible)?

Running anything from - https://github.com/kubernetes/community/blob/master/contributors/devel/sig-node/e2e-node-tests.md, such as:

```
make test-e2e-node REMOTE=true
```

### Anything else we need to know?

_No response_

### Kubernetes version

All

### Cloud provider

GCP (kind of)

### OS version

N/A

### Install tools

N/A

### Container runtime (CRI) and version (if applicable)

N/A

### Related plugins (CNI, CSI, ...) and versions (if applicable)

N/A

### PR DESCRIPTION
**Pull request by matthyx**
#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:
node e2e test installs docker from https://get.docker.com/ using curl, this is a supply chain security risk

#### Which issue(s) this PR fixes:

Fixes #109577

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by endocrimes** [comment]
@matthyx Hmm we probably actually want to update this to tell folks to install containerd + nerdctl now dockershim is gone

**Comment by matthyx** [comment]
> @matthyx Hmm we probably actually want to update this to tell folks to install containerd + nerdctl now dockershim is gone

But these appear to be for containerd, no?
https://github.com/containerd/containerd/blob/main/docs/getting-started.md#option-2-from-apt-get-or-dnf

Maybe I will split instructions for CentOS, Debian, Fedora and Ubuntu...
(now let's look at `nerdctl`)

**Comment by matthyx** [comment]
@endocrimes I have completely removed docker installation from the script, and now recommend containerd+nerdctl instead.
I don't know if we should autodetect the Linux distro and launch the instructions automagically, because it would add more toil to maintain, WDYT?

File: test/e2e_node/environment/setup_host.sh [hunk 1]
```diff
@@ -71,18 +67,8 @@ fi
 
 # Install docker
 if ! hash docker 2>/dev/null; then
```
Comment (endocrimes) [inline test/e2e_node/environment/setup_host.sh:69]:
```suggestion
if ! hash containerd 2>/dev/null; then
```

File: test/e2e_node/environment/setup_host.sh [hunk 2]
```diff
@@ -71,18 +67,8 @@ fi
...
-    sudo yum install -y docker-latest
-    sudo groupadd docker
-    sudo systemctl enable docker-latest.service
-    sudo systemctl start docker-latest.service
-  else
-    curl -fsSL https://get.docker.com/ | sh
-    sudo service docker start
-    sudo systemctl enable docker.service
-  fi
+  echo "Please install containerd.io using official packages, depending on your distro: https://docs.docker.com/engine/install/#server"
```
Comment (endocrimes) [inline test/e2e_node/environment/setup_host.sh:70]:
```suggestion
  echo "Please install containerd, see the getting started guide here: https://github.com/containerd/containerd/blob/main/docs/getting-started.md"
```

**Review by endocrimes — CHANGES_REQUESTED** [review]
almost lgtm

**Comment by matthyx** [comment]
Thanks @endocrimes done.

**Review by endocrimes — APPROVED** [review]
/lgtm
/approve

**Pull request merged**